packages/babel-plugin-polyfill-corejs2/src/index.ts

```diff
@@ -11,31 +9,33 @@ import addPlatformSpecificPolyfills from "./add-platform-specific-polyfills";
 import { hasMinVersion } from "./helpers";
 
 import defineProvider from "@babel/helper-define-polyfill-provider";
-import * as babel from "@babel/core";
-const { types: t } = babel.default || babel;
+import { types as t } from "@babel/core";
 
 const presetEnvCompat = "#__secret_key__@babel/preset-env__compatibility";
 const runtimeCompat = "#__secret_key__@babel/runtime__compatibility";
 
-// $FlowIgnore
 const has = Function.call.bind(Object.hasOwnProperty);
 
-type Options = {|
+type Options = {
   "#__secret_key__@babel/preset-env__compatibility": void | {
-    entryInjectRegenerator: boolean,
-  },
+    entryInjectRegenerator: boolean;
+  };
   "#__secret_key__@babel/runtime__compatibility": void | {
-    useBabelRuntime: string,
-    runtimeVersion: string,
-    ext: string,
-  },
-|};
+    useBabelRuntime: string;
+    runtimeVersion: string;
+    ext: string;
+  };
+};
 
 export default defineProvider<Options>(function (
   api,
   {
-    [presetEnvCompat]: { entryInjectRegenerator } = {},
-    [runtimeCompat]: { useBabelRuntime, runtimeVersion, ext = ".js" } = {},
+    [presetEnvCompat]: { entryInjectRegenerator = false } = {},
+    [runtimeCompat]: {
+      useBabelRuntime = "",
+      runtimeVersion = "",
```

Comment:
Those defaults are ok. However, if the object is provided it should already have all the properties and not rely on a default fallback. We should define it like this:

```js
[presetEnvCompat]: { entryInjectRegenerator } = { entryInjectRegenerator: false }
```

Reply:
Is there a functional difference between the two? not sure if I'm missing something or if that's just stylistic

Reply:
There is a difference when you pass an empty object:
```js
function a({ x } = { x: 2 }) { return x }
function b({ x = 2 } = {}) { return x }

a({}); // undefined
b({}); // 2
```

Reply:
In this specific case, I'm asking for the first behavior because you should either pass a complete object with all the necessary properties, or you don't pass the object. You cannot pass a "partial object", where some properties are specified and others have the default value.

Reply:
Ah I see, thanks for the clarification! The destructuring on destructuring syntax got a little confusing for me.

I'll convert it to this format shortly. Should I convert the `ext = ".js"` default as well? As that one was already there

Reply:
Converted to this format, but left `ext = ".js"` as that's what it was before. Let me know if `ext` should be changed as well

Reply:
See [follow-up discussion](https://github.com/babel/babel-polyfills/pull/121#discussion_r840851685)